src/rdb/rdb_raft.c

```diff
@@ -2289,39 +2330,14 @@ rdb_raft_load_lc(struct rdb *db)
 		goto err;
 	}
 
-lc:
-	/* Look up and open the LC. */
-	d_iov_set(&value, &db->d_lc_record, sizeof(db->d_lc_record));
-	rc = rdb_mc_lookup(db->d_mc, RDB_MC_ATTRS, &rdb_mc_lc, &value);
-	if (rc != 0) {
-		D_ERROR(DF_DB": failed to look up LC: "DF_RC"\n", DP_DB(db),
-			DP_RC(rc));
-		goto err_slc;
-	}
-	rc = vos_cont_open(db->d_pool, db->d_lc_record.dlr_uuid, &db->d_lc);
-	if (rc != 0) {
-		D_ERROR(DF_DB": failed to open LC "DF_UUID": %d\n", DP_DB(db),
-			DP_UUID(db->d_lc_record.dlr_uuid), rc);
-		goto err_slc;
-	}
-
-	/* Recover the LC by discarding any partially appended entries. */
-	rc = rdb_lc_discard(db->d_lc, db->d_lc_record.dlr_tail,
-			    RDB_LC_INDEX_MAX);
-	if (rc != 0) {
-		D_ERROR(DF_DB": failed to recover LC "DF_U64": %d\n", DP_DB(db),
-			db->d_lc_record.dlr_base, rc);
-		goto err_lc;
-	}
-
+load_snapshot:
 	/* Load the LC base. */
 	rc = rdb_raft_load_snapshot(db);
 	if (rc != 0)
-		goto err_lc;
+		goto err_slc;
 
 	/* Load the log entries. */
-	for (i = db->d_lc_record.dlr_base + 1; i < db->d_lc_record.dlr_tail;
-	     i++) {
+	for (i = db->d_lc_record.dlr_base + 1; i < db->d_lc_record.dlr_tail; i++) {
 		/*
 		 * Yield before loading the first entry (for the rdb_lc_discard
 		 * call above) and every a few entries.
```

Comment:
Good catch. I think an extra yield should not hurt since we are not on a performance-critical path. So I've remove the mention of rdb_lc_discard here and added a yield to rdb_raft_lc_open.